File: src/main/java/magasumovn/deanery/model/Performance.java
package magasumovn.deanery.model;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table
@IdClass(PerformanceCK.class)
public class Performance implements Serializable{
    @Id
    @ManyToOne()
    @JoinColumn(name = "studentID")
    private Student student;

    @Id
    @ManyToOne()
    @JoinColumn(name = "subjectID")
    private Subject subject;

    @Id
    private int semesterNumber;

    private int mark;

    public Student getStudent() {
        return student;
    }

    public Subject getSubject() {
        return subject;
    }

    public int getSemesterNumber() {
        return semesterNumber;
    }

    public int getMark() {
        return mark;
    }

    public void setMark(int mark) {
        this.mark = mark;
    }
}

File: src/main/resources/static/js/api/performance.js
import Vue from 'vue'

const performances = Vue.resource('/performance/{id}');

export default {
    get: studentID => performances.get({studentID}),
    getByMark: mark => performances.get({mark}),
    getByStudentName: studentName => performances.get({studentName}),
    save: performance => performances.save({}, performance),
    update: (performance) => performances.update({studentID: performance.student.studentID, subjectID: performance.subject.subjectID, semesterNumber: performance.semesterNumber}, performance),
    remove: (performance) => performances.remove({studentID: performance.student.studentID, subjectID: performance.subject.subjectID, semesterNumber: performance.semesterNumber}),
}